File: TestTemplateJS-Objc/TestTemplateJS-Objc/ViewController.m
//
//  ViewController.m
//  TestTemplateJS-Objc
//
//  Created by zhsoft88 on 2017/1/9.
//  Copyright © 2017年 zhuatang. All rights reserved.
//

#import "ViewController.h"
#import "ZHTHelperImpl.h"

@interface ViewController ()

@property (weak, nonatomic) IBOutlet UITextView *resultTextView;

@property (nonatomic) NSString* templateJS;
@property (nonatomic) NSString* dataJSON;

@end

@implementation ViewController

- (NSString *)templateJS {
  if (_templateJS == nil) {
    NSString* path = [[NSBundle mainBundle] pathForResource:@"template-1.0.min" ofType:@"js"];
    _templateJS = [[NSString alloc] initWithContentsOfFile:path encoding:NSUTF8StringEncoding error:nil];
  }
  
  return _templateJS;
}

- (NSString *)dataJSON {
  if (_dataJSON == nil) {
    NSString* path = [[NSBundle mainBundle] pathForResource:@"test-template-modal" ofType:@"json"];
    NSData* data = [NSData dataWithContentsOfFile:path];
    id object = [NSJSONSerialization JSONObjectWithData:data options:0 error:nil];
    // transform from human readable json
    NSData* output = [NSJSONSerialization dataWithJSONObject:object options:0 error:nil];
    _dataJSON = [[NSString alloc] initWithData:output encoding:NSUTF8StringEncoding];
  }
  
  return _dataJSON;
}

- (void)viewDidLoad {
  [super viewDidLoad];
  // Do any additional setup after loading the view, typically from a nib.
  JSContext* context = [JSContext new];
  [context evaluateScript:self.templateJS];
  context[@"helper"] = [ZHTHelperImpl new];
  NSString* script = [NSString stringWithFormat:@"var source = helper.getTemplateSourceById('test-template'); var tmpl = new Template(source); var data = %@; tmpl.render(data)", self.dataJSON];
  NSString* result = [[context evaluateScript:script] toString];
  self.resultTextView.text = result;
}


- (void)didReceiveMemoryWarning {
  [super didReceiveMemoryWarning];
  // Dispose of any resources that can be recreated.
}


@end

File: TestTemplateJS-Swift/TestTemplateJS-Swift/ViewController.swift
//
//  ViewController.swift
//  TestTemplateJS-Swift
//
//  Created by zhsoft88 on 2017/1/9.
//  Copyright © 2017年 zhuatang. All rights reserved.
//

import UIKit
import JavaScriptCore

class ViewController: UIViewController {

  @IBOutlet weak var resultTextView: UITextView!
  
  var templateJS : String {
    if let path = Bundle.main.path(forResource: "template-1.0.min", ofType: "js"),
      let contents = try? String(contentsOfFile: path) {
      return contents
    }
    
    return ""
  }
  
  var dataJSON : String {
    if let path = Bundle.main.path(forResource: "test-template-modal", ofType: "json"),
      let contents = try? String(contentsOfFile: path),
      let data = contents.data(using: String.Encoding.utf8),
      let obj = try? JSONSerialization.jsonObject(with: data, options: JSONSerialization.ReadingOptions()),
      // transform from human readable json
      let output = try? JSONSerialization.data(withJSONObject: obj, options: JSONSerialization.WritingOptions()),
      let result = String(data: output, encoding: String.Encoding.utf8) {
      return result
    }
    
    return ""
  }
  
  override func viewDidLoad() {
    super.viewDidLoad()
    // Do any additional setup after loading the view, typically from a nib.
    if let context = JSContext() {
      context.evaluateScript(templateJS)
      context.setObject(ZHTHelperImpl(), forKeyedSubscript: "helper" as (NSCopying & NSObjectProtocol)!)
      let script = String(format: "var source = helper.getTemplateSourceById('test-template'); var tmpl = new Template(source); var data = %@; tmpl.render(data)", dataJSON)
      let result = context.evaluateScript(script).toString()
      resultTextView.text = result
    }
    
  }

  override func didReceiveMemoryWarning() {
    super.didReceiveMemoryWarning()
    // Dispose of any resources that can be recreated.
  }


}

